src/Init/Data/Nat/Lemmas.lean

```diff
@@ -800,6 +801,53 @@ theorem mul_add_mod (m x y : Nat) : (m * x + y) % m = y % m := by
   · exact (m % 0).div_zero
   · case succ n => exact Nat.div_eq_of_lt (m.mod_lt n.succ_pos)
 
+theorem mod_eq_sub_mul_div (a b : Nat) : a % b = a - b * (a / b) := by
+  have := Nat.mod_add_div a b
+  have := Nat.mul_div_le a b
+  omega
+
+theorem lt_mul_div_self_add {x k : Nat} (h : 0 < k) : x < k * (x / k) + k := by
+  rw [← Int.ofNat_lt]
+  simpa using Int.lt_mul_ediv_self_add (by omega)
+
+theorem mul_sub_mod {x n p : Nat} (h : x < n * p) : (n * p - (x + 1)) % n = n - (x % n + 1) := by
```

Comment:
@fpvandoorn gave a different proof at https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Nat.2Emul_sub_mod/near/428107094; perhaps should refactor to use that.

Reply:
I've now included a shorter proof by Junyan Xu.